src/integration/Sites.spec.ts

```diff
@@ -0,0 +1,205 @@
+
+// Set up express with defaults and use the router under test
+const subrouter = express()
+// As we set certain properties on res.locals when the user signs in using github
+// In order to do integration testing, we must expose a middleware
+// that allows us to set this properties also
+subrouter.use((req, res, next) => {
+  const userSessionData = new UserSessionData({
+    isomerUserId: mockIsomerUserId,
+    email: mockEmail,
+  })
+  res.locals.userSessionData = userSessionData
+  next()
+})
+subrouter.use(sitesSubrouter)
+const app = generateRouter(subrouter)
+
+describe("Sites Router", () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+    mockAxios.reset()
+  })
+
+  describe("/", () => {
+    beforeAll(async () => {
+      // Set up User and Site table entries
+      await User.create({
+        id: mockIsomerUserId,
+      })
+      await Site.create({
+        id: mockSiteId,
+        name: mockSite,
+        apiTokenName: "token",
+        jobStatus: "READY",
+        siteStatus: "LAUNCHED",
+        creatorId: mockIsomerUserId,
+      })
+      await Site.create({
+        id: "200",
+        name: mockBlockedSite,
+        apiTokenName: "token",
+        jobStatus: "READY",
+        siteStatus: "LAUNCHED",
+        creatorId: mockIsomerUserId,
+      })
+      await SiteMember.create({
+        userId: mockIsomerUserId,
+        siteId: mockSiteId,
+        role: "ADMIN",
+      })
+      await Repo.create({
+        name: mockSite,
+        url: "url",
+        siteId: mockSiteId,
+      })
+    })
+    afterEach(async () => {
+      // Clean up so that different tests using
+      // the same mock user don't interfere with each other
+      await IsomerAdmin.destroy({
+        where: { userId: mockIsomerUserId },
+      })
+    })
+    it("should return list of only sites available to user", async () => {
+      // Arrange
+      const expected = {
+        siteNames: [
+          {
+            lastUpdated: mockUpdatedAt,
+            repoName: mockSite,
+            isPrivate: mockPrivate,
+            permissions: mockPermissions,
+          },
+        ],
+      }
+      ;((genericGitHubAxiosInstance.get as unknown) as jest.Mock).mockImplementationOnce(
+        () => ({
+          data: [
+            {
+              pushed_at: mockUpdatedAt,
+              permissions: mockPermissions,
+              name: mockSite,
+              private: mockPrivate,
+            },
+            {
+              pushed_at: mockUpdatedAt,
+              permissions: mockPermissions,
+              name: mockBlockedSite,
+              private: mockPrivate,
+            },
+          ],
+        })
+      )
+      ;((genericGitHubAxiosInstance.get as unknown) as jest.Mock).mockImplementationOnce(
+        () => ({
+          data: [],
+        })
+      )
+      ;((genericGitHubAxiosInstance.get as unknown) as jest.Mock).mockImplementationOnce(
+        () => ({
+          data: [],
+        })
+      )
```

Comment:
actually i think this works b/c there's a default `axios` mock in `__mocks__` - we should be using that instead HAHA

Reply:
I tried that earlier actually, but couldn't get it to work - I think it might have something to do with the `AxiosInstance` defined elsewhere which uses both `create` and interceptors. This was the way I got it to work with `genericGithubAxiosInstance`, but do let me know if it can be done with `mockAxios` instead!

Reply:
resolved this - i left a comment in the code for this but the tl;dr is that `SitesService` is hitting the `genericGithubAxiosInstance`, which is a named export. hence, this requires a mock of the named export after which everything will work fine